code/game/sound.dm

```diff
@@ -20,16 +20,18 @@
 	for(var/P in listeners)
 		var/mob/M = P
 		if(get_dist(M, turf_source) <= maxdistance)
-			sound_or_datum(M, turf_source, input, vol, vary, frequency, falloff, channel, pressure_affected)
+			sound_or_datum(M, turf_source, source, input, vol, vary, frequency, falloff, channel, pressure_affected)
 	for(var/P in SSmobs.dead_players_by_zlevel[z])
 		var/mob/M = P
 		if(get_dist(M, turf_source) <= maxdistance)
-			sound_or_datum(M, turf_source, input, vol, vary, frequency, falloff, channel, pressure_affected)
+			sound_or_datum(M, turf_source, source, input, vol, vary, frequency, falloff, channel, pressure_affected)
 
-/proc/sound_or_datum(mob/receiver, turf/turf_source, input, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE)
+/proc/sound_or_datum(mob/receiver, turf/turf_source, atom/source, input, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE)
```

Comment:
You need to add possibility of the source being a datum like with the case of squeak and move cooldown storage in the list inside the datum for such cases too

Reply:
squeak datums dont run their own sounds, the item they're linked to generates a datum then hands that down into the datum/component as an argument, also, if i gave squeak its own list wouldnt i have to make a datum subtype for every single plushie?

Reply:
In the end every plushie has a separate datum output to it based on type of plushie
Take that datum, put in into the plushies datum_outputs list as a key and value set to current time